### PR TITLE
pkg/clusteragent/admission: configure init resources

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

Library injection via Admission Controller: Allow configuring the CPU and Memory requests/limits for library init containers.

### Motivation

FR

### Describe how to test/QA your changes

Configure `admission_controller.auto_instrumentation.init_resources.cpu|memory` and ensures it gets applied to the init container.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by sgnn7 — APPROVED** [review]
LGTM for `pkg/config/config.go`

File: pkg/config/config.go [hunk 1]
```diff
@@ -1029,6 +1029,10 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("admission_controller.auto_instrumentation.patcher.enabled", false)
 	config.BindEnvAndSetDefault("admission_controller.auto_instrumentation.patcher.fallback_to_file_provider", false)                                // to be enabled only in e2e tests
 	config.BindEnvAndSetDefault("admission_controller.auto_instrumentation.patcher.file_provider_path", "/etc/datadog-agent/patch/auto-instru.json") // to be used only in e2e tests
+	config.BindEnv("admission_controller.auto_instrumentation.init_resources.cpu.request")
+	config.BindEnv("admission_controller.auto_instrumentation.init_resources.cpu.limit")
+	config.BindEnv("admission_controller.auto_instrumentation.init_resources.memory.request")
+	config.BindEnv("admission_controller.auto_instrumentation.init_resources.memory.limit")
```
Comment (clamoriniere) [inline pkg/config/config.go:1035]: maybe we can just set request==limit in that case. because it should be small enough

**Review by clamoriniere — COMMENTED** [review]
The change looks good.
but maybe for user config simplicity. default values should be provided.
It should be easy to check how much memory the process need to copy the files.

File: pkg/config/config.go [hunk 2]
```diff
@@ -1029,6 +1029,8 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("admission_controller.auto_instrumentation.patcher.enabled", false)
 	config.BindEnvAndSetDefault("admission_controller.auto_instrumentation.patcher.fallback_to_file_provider", false)                                // to be enabled only in e2e tests
 	config.BindEnvAndSetDefault("admission_controller.auto_instrumentation.patcher.file_provider_path", "/etc/datadog-agent/patch/auto-instru.json") // to be used only in e2e tests
+	config.BindEnv("admission_controller.auto_instrumentation.init_resources.cpu")
+	config.BindEnv("admission_controller.auto_instrumentation.init_resources.memory")
```
Comment (clamoriniere) [inline pkg/config/config.go:1033]: I think these new option should also be documented in https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml

**Review by kayayarai — APPROVED** [review]

File: pkg/clusteragent/admission/mutate/auto_instrumentation_test.go [hunk 3]
```diff
@@ -463,3 +464,80 @@ func TestInjectLibConfig(t *testing.T) {
...
+			name:    "with resources",
+			pod:     fakePod("java-pod"),
+			cpu:     "100m",
+			mem:     "500",
+			image:   "gcr.io/datadoghq/dd-lib-java-init:v1",
+			lang:    java,
+			wantErr: false,
+		},
+		{
+			name:    "with partial resources",
```
Comment (davidor) [inline pkg/clusteragent/admission/mutate/auto_instrumentation_test.go:495]: Should this test set only cpu or mem?

Comment (ahmed-mez) [reply pkg/clusteragent/admission/mutate/auto_instrumentation_test.go]: Updated accordingly! Thanks!

**Review by davidor — APPROVED** [review]

**Pull request merged**